src/php/Repl.php

```diff
@@ -8,106 +10,114 @@
         $this->output('Type "quit" or press Ctrl-D to exit.' . "\n");
+
         while (true) {
             $this->output("\e[?2004h"); // Enable bracketed paste
             $input = $this->readline->readline('>>> ');
             $this->output("\e[?2004l"); // Disable bracketed paste
-
-            if ($input === false) {
-                $this->output($this->color("Bye from Ctrl+D!\n", 'yellow'));
-                exit;
-            }
-
-            if ($input == 'quit') {
-                $this->output($this->color("Bye!\n", 'yellow'));
-                exit;
-            }
-
-            if ($input == '') {
-                continue;
-            }
-
-            $this->readline->addHistory($input);
-
-            try {
-                $tokenStream = $lexer->lexString($input);
-                $readerResult = $reader->readNext($tokenStream);
-
-                if (!$readerResult) {
-                    $this->output("Nothing to evaluate.\n");
-                    continue;
-                }
-
-                try {
-                    $node = $analzyer->analyze(
-                        $readerResult->getAst(),
-                        NodeEnvironment::empty()->withContext(NodeEnvironment::CTX_RET)
-                    );
-                    $code = $emitter->emitAsString($node);
-                    $result = $emitter->eval($code);
-
-                    $this->output($printer->print($result, false));
-                    $this->output(PHP_EOL);
-                } catch (AnalyzerException $e) {
-                    $exceptionPrinter->printException($e, $readerResult->getCodeSnippet());
-                }
-            } catch (ReaderException $e) {
-                $exceptionPrinter->printException($e, $e->getCodeSnippet());
-            } catch (Throwable $e) {
-                $exceptionPrinter->printStackTrace($e);
-            }
+            $this->readInput($input);
         }
     }
 
-    protected function color($text = '', $color = null)
+    private function output($value): void
+    {
+        fwrite(STDOUT, $value);
+    }
+
+    private function color($text = '', $color = null): string
     {
-        $styles = array(
-            'green'  => "\033[0;32m%s\033[0m",
-            'red'    => "\033[31;31m%s\033[0m",
+        $styles = [
+            'green' => "\033[0;32m%s\033[0m",
+            'red' => "\033[31;31m%s\033[0m",
             'yellow' => "\033[33;33m%s\033[0m",
-            'blue'   => "\033[33;34m%s\033[0m",
-        );
+            'blue' => "\033[33;34m%s\033[0m",
+        ];
 
         return sprintf($styles[$color] ?? "%s", $text);
     }
 
-    protected function input()
+    /**
+     * @param false|string $input
+     */
+    private function readInput($input): void
     {
-        return fgets(STDIN);
+        if (false === $input) {
+            $this->output($this->color("Bye from Ctrl+D!\n", 'yellow'));
+            exit;
+        }
+
+        if ('quit' === $input) {
```

Comment:
I would define not only `quit` but also `exit`.  What do you think, @jenshaase?
```php
if (in_array($input, ['quit', 'exit'], true)) { // extracting this array into a `priv const`
    $this->output($this->color("Bye!\n", 'yellow'));
    exit;
}
```

Reply:
I would go either for "quit" or for "exit". But you are right. "exit" is probably more common to PHP programmers, because of the "exit" function. @JesusValera Can you change "quit" to "exit"